OVB sensitivity - Cinelli and Hazlett(2020)

damandhaliwal opened the pull request:
Hi @s3alfisc,

Putting in a draft PR here for your eyes. Let me know if you have any feedback on the API setup.

Here's what's left (I'm on it):
- Add documentation
- Benchmarking
- Visualizations
- Clean outputs/reporting

Just wanted to see if you had any bigger API design level changes before I went down the rabbit hole and implemented everything

damandhaliwal committed:
partial r2 done. To be finished tomorrow and then tested

damandhaliwal committed:
rv done. summary stats and test left before PR

damandhaliwal committed:
initial draft for the PR.

damandhaliwal committed:
basic test to check it works

damandhaliwal committed:
benchmarking complete

damandhaliwal committed:
completed summary function and added documentation.

damandhaliwal committed:
merge branch 'master' into ovb_sensitivity

s3alfisc commented on pyfixest/estimation/sensitivity.py:
```diff
@@ -0,0 +1,472 @@
+        pd.DataFrame
+            DataFrame with 'r2dz_x' and 'r2yz_dx' columns.
+        """
+        from pyfixest.estimation.estimation import feols
```
This path has changed in the main branch, you know have to load as `from pyfixest.estimation.api import feols`

s3alfisc commented:
Could you add some defensive checks (plus tests in `test_errors_warnigns`) that throw an error when users try to use the sensitivity method for models other than the OLS class? Add a `self._supports_sensitivity_analysis = True` for `Feols`, but set it to `False` in the `__init__` of the other model classes? Else all other model classes will inherit the sensitivity method from `Feols` and incorrectly run the analysis?

s3alfisc commented:
Last - it would be great to write a "how to" notebook - what cam you do with the method, why is it needed, how does it work, maybe why the authors argue it is better than other methods (as e.g. Oster)? Maybe [something like this](https://cran.r-project.org/web/packages/sensemakr/vignettes/sensemakr.html), but even more to the point?

s3alfisc committed:
Merge remote-tracking branch 'origin/master' into ovb_sensitivity
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

s3alfisc committed:
move from estimation.estimation to estimation.api

damandhaliwal committed:
Merge branch 'ovb_sensitivity' of github.com:damandhaliwal/pyfixest into ovb_sensitivity

damandhaliwal committed:
Merge remote-tracking branch 'upstream/master' into ovb_sensitivity

damandhaliwal committed:
self._supports_sensitivity added

damandhaliwal committed:
add checks and tests for other methods

damandhaliwal committed:
plotting functions added. Tests for these pending

damandhaliwal committed:
fix missing treatment param in summary() and add plotting tests

- Add treatment parameter to SensitivityAnalysis.summary() method
- Fix unbound plot_estimate variable in ovb_contour_plot for t-value sensitivity
- Add 10 tests for contour and extreme plotting functions

damandhaliwal commented:
> Last - it would be great to write a "how to" notebook - what cam you do with the method, why is it needed, how does it work, maybe why the authors argue it is better than other methods (as e.g. Oster)? Maybe [something like this](https://cran.r-project.org/web/packages/sensemakr/vignettes/sensemakr.html), but even more to the point?

I have added everything but this. I think I will add this in a separate PR. I can draft an issue for this once we finalize and merge this PR

s3alfisc committed:
ruff, mypy

s3alfisc committed:
align with main
